packages/grafana-ui/src/components/FileUpload/FileUpload.tsx

```diff
@@ -1,39 +1,45 @@
 import React, { FC, FormEvent } from 'react';
-import { getFormStyles, stylesFactory, useTheme } from '@grafana/ui';
 import { GrafanaTheme } from '@grafana/data';
-import { css } from 'emotion';
+import { css, cx } from 'emotion';
+import { getFormStyles } from '../index';
+import { stylesFactory, useTheme } from '../../themes';
 
-interface Props {
+export interface Props {
   onFileUpload: (event: FormEvent<HTMLInputElement>) => void;
+  /** Custom upload label text */
+  label?: string;
+  /** Accepted file extensions */
+  accept?: string;
+  className?: string;
 }
 
-const getStyles = stylesFactory((theme: GrafanaTheme) => {
-  const buttonFormStyle = getFormStyles(theme, { variant: 'primary', invalid: false, size: 'md' }).button.button;
-  return {
-    fileUpload: css`
-      display: none;
-    `,
-    button: css`
-      ${buttonFormStyle}
-    `,
-  };
-});
-
-export const DashboardFileUpload: FC<Props> = ({ onFileUpload }) => {
+export const FileUpload: FC<Props> = ({ onFileUpload, className, label = 'Upload file', accept = '*' }) => {
   const theme = useTheme();
   const style = getStyles(theme);
 
   return (
-    <label className={style.button}>
-      Upload .json file
+    <label className={cx(style.button, className)}>
+      {label}
```

Comment:
This might be an idea
```suggestion
      <Icon name="upload"/>
      {label}
```

Reply:
Good idea!